crates/plugins/src/lookup.rs

```diff
@@ -30,7 +30,38 @@ impl PluginLookup {
         }
     }
 
-    pub async fn get_manifest_from_repository(
+    pub async fn resolve_manifest(
+        &self,
+        plugins_dir: &Path,
+        skip_compatibility_check: bool,
+        spin_version: &str,
+    ) -> PluginLookupResult<PluginManifest> {
+        let exact = self.resolve_manifest_exact(plugins_dir).await?;
+        if skip_compatibility_check
+            || self.version.is_some()
+            || exact.is_compatible_spin_version(spin_version)
+        {
+            Ok(exact)
+        } else {
+            let store = crate::store::PluginStore::new(plugins_dir.to_owned());
+
+            // This is very similar to some logic in the badger module
+            let manifests = store.catalogue_manifests()?;
+            let relevant_manifests = manifests.into_iter().filter(|m| m.name() == self.name);
+            let compatible_manifests = relevant_manifests.filter(|m| {
+                m.has_compatible_package() && m.is_compatible_spin_version(spin_version)
+            });
+            let highest_compatible_manifest =
+                compatible_manifests.max_by_key(|m| semver::Version::parse(&m.version).unwrap());
```

Comment:
Is unwrapping the right thing to do here? Can we do `unwrap_or_else(|| semver::Version::new(0,0,0))`?